### PR TITLE
[rush] Add support for RUSH_PREVIEW_VERSION environment variable

### PR DESCRIPTION
**Pull request by pgonzal**
To test out a new Rush version, developers can now set an environment variable RUSH_PREVIEW_VERSION.

Rush will display a warning banner like this:

```
* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
  WARNING! THE "RUSH_PREVIEW_VERSION" ENVIRONMENT VARIABLE IS SET.

  You are previewing Rush version:        5.0.0-dev.16
  The rush.json configuration asks for:   5.0.0-dev.17

  To restore the normal behavior, unset the RUSH_PREVIEW_VERSION
  environment variable.
* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
```

### CODE REVIEW

File: apps/rush/src/start.ts [hunk 1]
```diff
@@ -35,44 +36,68 @@ else if (!semver.satisfies(nodeVersion, '^6.9.0')
...
-      currentPackageJson.version,
-      isManaged // Rush is "managed" if its version and configuration are dictated by a repo's rush.json
-    );
+let rushVersionToLoad: string | undefined = undefined;
+
+const previewVersion: string | undefined = process.env[EnvironmentVariableNames.RUSH_PREVIEW_VERSION];
+
+if (previewVersion) {
+  if (!semver.valid(previewVersion, false)) {
+    console.error(colors.red(`Invalid value for RUSH_PREVIEW_VERSION: "${previewVersion}"`));
```
Comment (iclanton) [inline apps/rush/src/start.ts]: >RUSH_PREVIEW_VERSION [](start = 48, length = 20)

Explicitly mention that this is an env variable.

File: apps/rush/src/start.ts [hunk 2]
```diff
@@ -35,44 +36,68 @@ else if (!semver.satisfies(nodeVersion, '^6.9.0')
...
+    process.exit(1);
```
Comment (iclanton) [inline apps/rush/src/start.ts]: >process.exit(1); [](start = 4, length = 16)

Can you put an explicit return?

File: apps/rush/src/start.ts [hunk 3]
```diff
@@ -35,44 +36,68 @@ else if (!semver.satisfies(nodeVersion, '^6.9.0')
...
+  }
+
+  rushVersionToLoad = previewVersion;
+
+  console.error(colors.yellow(
```
Comment (iclanton) [inline apps/rush/src/start.ts]: >colors [](start = 16, length = 6)

What about yellow background with black text?

File: apps/rush/src/start.ts (hunk 2 above)
Comment (pgonzal) [reply apps/rush/src/start.ts]: Returns "never"

---
In reply to: [187754990](https://github.com/Microsoft/web-build-tools/pull/653#discussion_r187754990) [](ancestors = 187754990)

**Pull request merged**